### PR TITLE
feat(dws): add workload plan execution resource support

### PR DESCRIPTION
**Pull request by jinyangyang222**
**What this PR does / why we need it**:
Use this resource to implement the function of starting, stopping and switching plan stage for workload plan resource
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
add workload plan execution resource support
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST="./huaweicloud/services/acceptance/dws" TESTARGS="-run TestAccResourceWorkLoadPlanExecution_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dws -v -run TestAccResourceWorkLoadPlanExecution_basic -timeout 360m -parallel 4
=== RUN   TestAccResourceWorkLoadPlanExecution_basic
=== PAUSE TestAccResourceWorkLoadPlanExecution_basic
=== CONT  TestAccResourceWorkLoadPlanExecution_basic
--- PASS: TestAccResourceWorkLoadPlanExecution_basic (1385.68s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dws       1385.720s
```

### CODE REVIEW

File: huaweicloud/services/acceptance/dws/resource_huaweicloud_dws_workload_plan_attachment_test.go [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+	}
+
+	getResp, err := client.Request("GET", getPath, &getOpt)
+	if err != nil {
+		return nil, fmt.Errorf("error querying DWS workload plan: %s", err)
+	}
+
+	getRespBody, err := utils.FlattenResponse(getResp)
+	if err != nil {
+		return nil, fmt.Errorf("%s", err)
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/dws/resource_huaweicloud_dws_workload_plan_attachment_test.go]:
```suggestion
		return nil, err
```

File: huaweicloud/services/acceptance/dws/resource_huaweicloud_dws_workload_plan_attachment_test.go [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+	}
+
+	// When calling the API, there is a situation where the plan ID does not exist but still returns a 200 status code.
+	// When the API is successfully called to start the workload plan, the value of the status attribute is 1.
+	errCode := utils.PathSearch("error_code", getRespBody, "").(string)
+	plan := utils.PathSearch("workload_plan", getRespBody, nil)
+	status := utils.PathSearch("status", plan, nil).(float64)
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/dws/resource_huaweicloud_dws_workload_plan_attachment_test.go]: This line may cause panic error.

File: huaweicloud/services/acceptance/dws/resource_huaweicloud_dws_workload_plan_attachment_test.go [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+func testAccWorkLoadPlanAttachment_basic(name string) string {
+	return fmt.Sprintf(`
+%s
+
+resource "huaweicloud_dws_workload_plan_attachment" "test" {
+  depends_on = [
+    huaweicloud_dws_workload_plan_stage.test1,
+    huaweicloud_dws_workload_plan_stage.test2,
+  ]
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/dws/resource_huaweicloud_dws_workload_plan_attachment_test.go]: I think the depends_on is unnecessary.

File: huaweicloud/services/dws/resource_huaweicloud_dws_workload_plan_attachment.go [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+
+	getRespBody, err := utils.FlattenResponse(getResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	// When calling the API, there is a situation where the plan ID does not exist but still returns a 200 status code.
+	// When the API is successfully called to start the workload plan, the value of the status attribute is 1.
+	errCode := utils.PathSearch("error_code", getRespBody, "").(string)
+	status := utils.PathSearch("workload_plan.status", getRespBody, nil).(float64)
```
Comment (deer-hang) [inline huaweicloud/services/dws/resource_huaweicloud_dws_workload_plan_attachment.go]: ditto

File: huaweicloud/services/acceptance/dws/resource_huaweicloud_dws_workload_plan_attachment_test.go (hunk 1 above)
Comment (jinyangyang222) [reply huaweicloud/services/acceptance/dws/resource_huaweicloud_dws_workload_plan_attachment_test.go]: fixed.

File: huaweicloud/services/acceptance/dws/resource_huaweicloud_dws_workload_plan_attachment_test.go (hunk 3 above)
Comment (jinyangyang222) [reply huaweicloud/services/acceptance/dws/resource_huaweicloud_dws_workload_plan_attachment_test.go]: fixed.

File: huaweicloud/services/acceptance/dws/resource_huaweicloud_dws_workload_plan_attachment_test.go (hunk 2 above)
Comment (jinyangyang222) [reply huaweicloud/services/acceptance/dws/resource_huaweicloud_dws_workload_plan_attachment_test.go]: fixed.

File: huaweicloud/services/dws/resource_huaweicloud_dws_workload_plan_attachment.go (hunk 4 above)
Comment (jinyangyang222) [reply huaweicloud/services/dws/resource_huaweicloud_dws_workload_plan_attachment.go]: fixed.

**Comment by jinyangyang222** [comment]
make testacc TEST="./huaweicloud/services/acceptance/dws" TESTARGS="-run TestAccResourceWorkLoadPlanExecution_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dws -v -run TestAccResourceWorkLoadPlanExecution_basic -timeout 360m -parallel 4
=== RUN   TestAccResourceWorkLoadPlanExecution_basic
=== PAUSE TestAccResourceWorkLoadPlanExecution_basic
=== CONT  TestAccResourceWorkLoadPlanExecution_basic
--- PASS: TestAccResourceWorkLoadPlanExecution_basic (1300.14s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dws       1300.180s

**Pull request merged**